File: 1/04.go
package main

import "fmt"
import "strings"

func is_in_ns(n int) bool {
	ns := []int{1, 5, 6, 7, 8, 9, 15, 16, 19}
	for _, e := range ns {
		if e == n {
			return true
		}
	}
	return false
}

func main() {
	s := "Hi He Lied Because Boron Could Not Oxidize Fluorine. New Nations Might Also Sign Peace Security Clause. Arthur King Can."
	a := strings.Fields(s)
	r := make(map[string]int)
	for i, w := range a {
		n := i + 1
		if is_in_ns(n) {
			r[w[0:1]] = n
		} else {
			r[w[0:2]] = n
		}
	}
	fmt.Println(r)
}

File: 1/04.pl
#!/usr/bin/perl
use strict;
use warnings;
my $s = "Hi He Lied Because Boron Could Not Oxidize Fluorine. New Nations Might Also Sign Peace Security Clause. Arthur King Can.";
my @a = split(/\s+/, $s);
my %h;
while (my ($i, $w) = each @a) {
  my $n = $i + 1;
  if (grep {$_ == $n} (1, 5, 6, 7, 8, 9, 15, 16, 19)) {
    $h{substr($w, 0, 1)} = $n;
  } else {
    $h{substr($w, 0, 2)} = $n;
  }
}
for my $k (sort {$h{$a} <=> $h{$b}} keys %h) {
  print "$k: $h{$k}\n";
}
